src/cli/commands/add.js

```diff
@@ -83,7 +83,7 @@ export class Add extends Install {
       let prefix;
       if (tilde) {
         prefix = '~';
-      } else if (exact) {
+      } else if (exact || Boolean(this.config.getOption('save-exact'))) {
```

Comment:
I think it is better to do
```
const exact = this.flags.exact || Boolean(this.config.getOption('save-exact'));
const tilde = this.flags.tilde;
```

above. `save-exact` should take precedence over `save-prefix` option.